java/src/org/openqa/selenium/firefox/AddHasExtensions.java

```diff
@@ -77,7 +84,23 @@ public String installExtension(Path path, Boolean temporary) {
 
         String encoded;
         try {
-          encoded = Base64.getEncoder().encodeToString(Files.readAllBytes(path));
+          if (Files.isDirectory(path)) {
+            Path extZip = Paths.get(path.getFileName().toString()+".zip");
+            ZipOutputStream zos = new ZipOutputStream(new FileOutputStream(extZip.toFile()));
+            Files.walkFileTree(path, new SimpleFileVisitor<Path>() {
```

Comment:
It should be fine if Path is not passed explicitly.

Reply:
Removing `Path` is breaking the code. I'll look into it further.